module-2/lab-subsetting-and-descriptive-stats/your-code/main.ipynb

```diff
@@ -21,7 +21,7 @@
   },
```

Comment:
- Find the mean and the standard deviation of the temperature of each state: another way without losing the object type (df)
```Python 
temp.groupby("State").agg({'Temperature':['mean','std']}) 
```
- Show visually the frequency distribution (histogram) of the employee dataset. In few words describe these histograms?: It's a good practice to have the number of bins under control, a high number of bins makes the histogram more difficult to interpret. It depends on the datatype and the length of the data set. We have a data set with only 9 records so the interpretation is not very useful. The most frequent seniority in the company is 2 years. The years distribution is an asymmetric distribution and the salary distribution is nearly symmetric. 
We can see that half of the employees are in the low salary band and nearly the other half in the high salary band. We don't have intermediate salary levels in our data set. 
- Print the first 3 rows of your dataframe: the head method is another way of getting the first 3 rows. 
- Find the employees who's title is associate and the salary above 55? You have use 50 in conditional.....
- Show a visual summary of the data using boxplot. What Are the First and Third Quartiles? Comment your results: box-plots are usually built for every numeric variable, in this case both Salary and Year. You've missed the interpretation. 
- It is useful to skip useless columns when you read a csv, in this case the unnamed column is useless and skiping it is a good choice. 
- Show the frequency distribution of the amount spent in Spain: the same as above be careful with the number of bins you use. 
- Bonus Question: For each department, compute the difference between the maximal salary and the minimal salary.Hint: try using agg or apply and lambda. Another way of doing this using agg or apply would be: 
```Python 
employee_stats = employee.groupby('Department', as_index=False).agg({'Salary': lambda x: max(x)-min(x)})
```
-It could have been a good idea to change the datatype of InvoiceDate to datetime or split it into 2 columns, date and time.